src/writer/vote.ts

```diff
@@ -176,7 +176,11 @@ export async function action(body, ipfs, receipt, id, context): Promise<void> {
     );
   } else {
     // Store vote in dedicated table
-    await db.queryAsync('INSERT IGNORE INTO votes SET ?', params);
+    const result = await db.queryAsync('INSERT IGNORE INTO votes SET ?', params);
+
+    if (result.affectedRows > 0) {
+      await incrementVotesCount(msg.space, voter);
```

Comment:
What if this fails due to some network issue after inserting?

Reply:
If first insert query succeed, then something happens, and vote increment fail, then the counter will not increment.

As we're not using query transaction, we're not protected against this use case.

Reply:
We can run cron to refresh some counters with the full computation (like in the generate scripts), instead of `+1`, to resync the counters

There's another way to keep the counters in sync, by using SQL triggers.

As for the solution used in keycard, it'll work only if we're 100% that the first query do insert in all cases, which does not seem the case, since it's using `INSERT IGNORE `, which imply that in some cases, it'll succeed, but not insert anything. In that case, we should not run the second query. Unless we're using a more complex SQL, with condition etc ...